ompi/instance/instance.c

```diff
@@ -498,6 +498,10 @@ static int ompi_mpi_instance_init_common (int argc, char **argv)
 
     /* Select which MPI components to use */
 
+    if (OPAL_SUCCESS != (ret = mca_smsc_base_select())) {
+        return ompi_instance_print_error ("mca_smsc_base_select() failed", ret);
```

Comment:
At the moment, mca_smsc_base_select() only returns OPAL_SUCCESS. The actual smsc components might still remain silently uninitialized. My though process here is that if it at some point in the future mca_smsc_base_select started possibly to returning something else, it would something severe enough that would warrant showing an error message.